apis/management.cattle.io/v3/cluster_types.go

```diff
@@ -329,3 +332,10 @@ type SaveAsTemplateOutput struct {
 	ClusterTemplateName         string `json:"clusterTemplateName,omitempty"`
 	ClusterTemplateRevisionName string `json:"clusterTemplateRevisionName,omitempty"`
 }
+
+type EKSStatus struct {
+	GeneratedVirtualNetwork string   `json:"generatedVirtualNetwork"`
+	GeneratedSubnets        []string `json:"generatedSubnets"`
+	GeneratedSecurityGroups []string `json:"generatedSecurityGroups"`
+	GeneratedServiceRole    string   `json:"generatedServiceRole"`
+}
```

Comment:
Wondering if this information can be generic and apply to other kind of clusters as well later? So instead of EKSStatus, maybe these fields can be carried together in a generic wrapper like "NetworkStatus" or similar

Reply:
I think these are subject to change and differ between provider, it makes sense to me for them to be in their own struct, if we end up doing something in the future that overlaps then I think we can revisit this. As it is right now, it is very easy for me to know which fields are specific to the EKS implementation and they match a subset of fields that appear in EKSClusterConfig status.